src/generic/loading-button/LoadingButton.test.jsx

```diff
@@ -0,0 +1,68 @@
+
+import LoadingButton from '.';
+
+const buttonTitle = 'Button Title';
+
+const RootWrapper = (onClick) => (
+  <LoadingButton label={buttonTitle} onClick={onClick} />
+);
+
+describe('<LoadingButton />', () => {
+  it('renders the title and doesnt handle the spinner initially', () => {
+    const { container, getByText } = render(RootWrapper(() => { }));
+    expect(getByText(buttonTitle)).toBeInTheDocument();
+
+    expect(container.getElementsByClassName('icon-spin').length).toBe(0);
+  });
+
+  it('doesnt render the spinner without onClick function', () => {
+    const { container, getByRole, getByText } = render(RootWrapper());
+    const titleElement = getByText(buttonTitle);
+    expect(titleElement).toBeInTheDocument();
+    expect(container.getElementsByClassName('icon-spin').length).toBe(0);
+    fireEvent.click(getByRole('button'));
+    expect(container.getElementsByClassName('icon-spin').length).toBe(0);
+  });
+
+  it('renders the spinner correctly', async () => {
+    const longFunction = () => new Promise((resolve) => {
+      setTimeout(resolve, 1000);
+    });
+    const { container, getByRole, getByText } = render(RootWrapper(longFunction));
+    const buttonElement = getByRole('button');
+    fireEvent.click(buttonElement);
+    expect(container.getElementsByClassName('icon-spin').length).toBe(1);
+    expect(getByText(buttonTitle)).toBeInTheDocument();
+    // StatefulButton only sets aria-disabled (not disabled) when the state is pending
+    // expect(buttonElement).toBeDisabled();
+    expect(buttonElement).toHaveAttribute('aria-disabled', 'true');
+    await waitFor(() => {
+      // StatefulButton only sets aria-disabled (not disabled) when the state is pending
+      // expect(buttonElement).toBeEnabled();
+      expect(buttonElement).not.toHaveAttribute('aria-disabled', 'true');
+      expect(container.getElementsByClassName('icon-spin').length).toBe(0);
+    });
+  });
+
+  it('renders the spinner correctly even with error', async () => {
+    const longFunction = () => new Promise((_resolve, reject) => {
+      setTimeout(reject, 1000);
+    });
+    const { container, getByRole, getByText } = render(RootWrapper(longFunction));
+    const buttonElement = getByRole('button');
+    fireEvent.click(buttonElement);
+    expect(container.getElementsByClassName('icon-spin').length).toBe(1);
+    expect(getByText(buttonTitle)).toBeInTheDocument();
+    // StatefulButton only sets aria-disabled (not disabled) when the state is pending
+    // expect(buttonElement).toBeDisabled();
+    expect(buttonElement).toHaveAttribute('aria-disabled', 'true');
+    await waitFor(() => {
+      // StatefulButton only sets aria-disabled (not disabled) when the state is pending
+      // expect(buttonElement).toBeEnabled();
+      expect(buttonElement).not.toHaveAttribute('aria-disabled', 'true');
+      expect(container.getElementsByClassName('icon-spin').length).toBe(0);
+    });
```

Comment:
This test seems to be throwing an error.

Reply:
Fixed here: f0af395d0b6b46131e330f49c534e7c55901b852